### ISSUE TITLE
What is the accuracy of PoseNet?

### ISSUE THREAD

**Issue by joyccino** [tensorflow/tensorflow#46728]
Is anyone here tried re-train posenet model?
I recognised there are no official codes to re-train a posenet model,
so I'm going to make a TF based pose-estimation with COCO and then convert it to TFLite style.
If there is anyone else already tried this, how did you guys compare the accuracy of the new posenet with the official one?
Because I could not find the accuracy of the pre-trained PoseNet model from official blogs or papers.
Also, I would like to try to configure the model to see how the model accuracy increases, but I have the same question here: how do I check the accuracy of the model to compare with the original one??

**Comment by srjoglekar246** [comment]
Yeah I am not sure if there is any published accuracy documentation of the model we host. Your plan to train your own sounds good, hopefully ensure that the ops you use are similar to what the [current model](https://tfhub.dev/tensorflow/lite-model/posenet/mobilenet/float/075/1/default/1) uses (try visualizing with Netron if needed); this will ensure seamless conversion to TFLite.

**Comment by joyccino** [comment]
Thanks srjoglekar246 for your quick response.
I wanted to try dropping the loss (and hopefully increase the accuracy) of the current model bc while adapting the model to my project, it seemed like the model has some difficulties with detecting a simple pose.
I tried changing the confidence score of each keypoints from 0.5 to 0.9, but the result was not that much satisfying.
So thought re-training (or adapting transfer-learning to) current model might give back some answer, however there are no available current accuracy and loss of the model that I can compare with... so that sounds like re-training the model might not be a meaningful next step for me.
Do you guys have any recommendation to increase the model accuracy (and decrease the loss) ?

**Comment by joyccino** [comment]
Thanks srjoglekar246 for your immediate response.
I wanted to try dropping the loss (and hopefully increase the accuracy) of the current model bc while adapting the model to my project, it seemed like the model could not detect many poses right (sometimes it gives a bit messed up results for a simple pose...).
So I tried changing the confidence score of each keypoints from 0.5 to 0.9, but the result was not that much satisfying.
I thought re-training (or adapting transfer-learning to) the current model might give back some pleasing answers, however as you said there are no available current accuracy and loss of the model that I can compare with... so I guess re-training the model might not be a meaningful next step for me.
Do you have any recommendation to increase the model accuracy (and decrease the loss) ?

**Comment by srjoglekar246** [comment]
@joyccino First thing, we can check if the way you are running inference with the PoseNet model aligns with what the model expects. [This Medium post](https://medium.com/roonyx/pose-estimation-and-matching-with-tensorflow-lite-posenet-model-ea2e9249abbd) shows some of the details in how to generate inputs & process outputs from a TFLite PostNet model. Could you check if you are doing the same logic in your code? Maybe also try with the same model that the author uses, for a fair comparison.

**Comment by jvishnuvardhan** [comment]
@joyccino Is this still an issue for you. I see a page on Pose estimation that has benchmark performance for different datasets. https://www.tensorflow.org/lite/examples/pose_estimation/overview

Thanks!
